x-pack/plugins/ml/public/application/data_frame_analytics/pages/job_map/components/controls.tsx

```diff
@@ -94,27 +159,94 @@ export const Controls: FC<Props> = ({ analyticsId, modelId, details, getNodeData
     };
   }, [cy, deselect]);
 
+  useEffect(() => {
+    function updateElementsOnClose() {
+      if (isModalVisible === false && deleteItem === true) {
+        let destIndexNode;
+        if (deleteTargetIndex === true) {
+          const jobDetails = details[nodeId];
+          const destIndex = jobDetails.dest.index;
+          destIndexNode = `${destIndex}-${JOB_MAP_NODE_TYPES.INDEX}`;
+        }
+        updateElements(nodeId, nodeLabel, destIndexNode);
+        setShowFlyout(false);
+      }
+    }
+
+    updateElementsOnClose();
+  }, [isModalVisible, deleteItem]);
```

Comment:
you should pass a named function directly to useEffect 
```suggestion
  useEffect(function updateElementsOnClose() {
      if (isModalVisible === false && deleteItem === true) {
        let destIndexNode;
        if (deleteTargetIndex === true) {
          const jobDetails = details[nodeId];
          const destIndex = jobDetails.dest.index;
          destIndexNode = `${destIndex}-${JOB_MAP_NODE_TYPES.INDEX}`;
        }
        updateElements(nodeId, nodeLabel, destIndexNode);
        setShowFlyout(false);
      }
  }, [isModalVisible, deleteItem]);
```